src/main/java/uk/gov/ch/controller/update/EntityDataController.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.ch.controller.update;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.ch.OracleQueryApplication;
+import uk.gov.ch.exception.EntityEmailNotFoundException;
+import uk.gov.ch.model.update.EntityData;
+import uk.gov.ch.update.impl.EntityDataServiceImpl;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@RestController
+public class EntityDataController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private EntityDataServiceImpl entityDataService;
+
+    @GetMapping("/company/{incorporationNumber}/update/entityData")
+    public ResponseEntity<EntityData> getEntityEmail(@PathVariable String incorporationNumber) {
+        try {
+            return ResponseEntity.ok(entityDataService.getEntityEmail(incorporationNumber));
```

Comment:
Need to create a Response object so that we return "{ email_address: <value> }" json.

Reply:
ok see this returns not just email address so maybe the getter is miss named?

Reply:
@AndreKramer7654 Yeah I had intended to do just that, but everywhere else in the app seems to use the Entity model (which I had always thought was bad practice, yet was consistent with what the app is already doing). Wanted to see what you thought about it. I can create a DTO.

Reply:
think we need separate DTO. Maybe the dto and the controller should be named OverseasEntityXYZ to make it clearer what they are about.